### DISCUSSION TITLE
Where is the code that dispatch the event "before_login_post_redirect"?

### DISCUSSION THREAD

**Discussion by kiatng** [OpenMage/magento-lts#3745] [Questions & Answers]
Ref 
https://github.com/OpenMage/magento-lts/blob/e2e738a884f24e09e05e7ad098946ac4d7ff877d/app/code/core/Mage/Oauth/etc/config.xml#L63-L70

I cannot find where is it dispatched:
https://github.com/search?q=repo%3AOpenMage%2Fmagento-lts+before_login_post_redirect&type=code

Any idea?

**Comment by sreichel** [comment]
Cant find it. Is it dispatched at all? Have you tried to set a xdebug breakpoint?

**Reply by kiatng** [reply]
Good suggestion. I added debug info, the event never dispatched in both login and OAuth customer login. I think it is a bug. OAuth login has 2 options: login as a customer or admin user. I am doing something to login as customer. When doing the OAuth customer login, the page is redirected to frontend My Account (as set in the configuration), but the expected behavior should be redirected to the callback URL specified by the OAuth consumer. I'll test some more and make a PR to dispatch the event.

**Comment by kiatng** [comment]
I am still trying to figure out the OAuth mechanism. 

1. Was the event `before_login_post_redirect` deprecated? [edit] No, the event is required to redirect customer to the authorize page after login.
2. After the customer login at OAuth host, where is the page redirected? Is it to the callback URL or the authorize page at the host? [edit] After login, the page should be redirected to the authorize page.
3. What happen if the customer choose to reject on the login page at the OAuth host? Should the page be redirected back to the OAuth consumer? [edit] If no callback URL if provided, `reject.phtml` is rendered; else it is redirected to the callback URL specified in the backend consumer config `$consumer->getRejectedCallbackUrl()` or in the param 'oauth_callback' specified by the consumer. The host also attaches a param `rejected=1` in the URL.
